### ISSUE TITLE
Instal memcached on windows

### ISSUE THREAD

**Issue by morious** [3rd-Eden/node-memcached#233]
Hi,
Is there any way to avoid the build during installation of memcached on windows?

Thx!

**Comment by 3rd-Eden** [comment]
No, because the hashring requires 64bit bitshifthing which is not possible in JavaScript and therefor requires a binary add-on so we can correctly distribute the servers in a cluster. We could make it an optional dependency but it means that when it fails to build you can only use one server instead of multiple servers and I don't think that this makes sense.

**Comment by morious** [comment]
Why can't the binary be pre-built and distributed with the package?

**Comment by 3rd-Eden** [comment]
Because it would have to be pre-build for every single operating system and node.js version combination and bundled within the package. So instead of downloading a 2mb file you start downloading a 200mb file because it's all bundled in. That's not really a solution. There is however a project called node-pre-gyp which makes it possible to create pre-build binaries and they are then stored on s3 but this is hosting you have to pay for your self and quit hard to setup and not really worth the time as I beleive that these are issues that npm needs resolve, not the module authors.

**Comment by dmitrymatveev** [comment]
I just use VirtualBox running a linux distro with hassle free memcached server during development on my windows machine. Works wonders

**Comment by morious** [comment]
@3rd-Eden first, there has got to be a way to install an architecture or os specific binary through npm rather than downloading 200mb of binaries, no? Second, how would you recommend approaching a server deployment of an application that has a mem-cached dependency?
Thx!

**Comment by 3rd-Eden** [comment]
@morious I'm just hoping that npm solves this problem. They should just pre-compile these kind of binary dependencies on their server and use those during installation. But I highly doubt that's something they are willing pursue. The `node-gyp-pre` module is a good initiative except for the part when you have the host the files your self and it's quite (read super) painful to setup.

Ideally, someone should write a module which just publishes the compiled binaries to npm with the suffix of the build info like `memcached-<namespace>-<os fluff>` and then magically downloads them using an `install` script in the `package.json`.

**Comment by morious** [comment]
@3rd-Eden I agree with both your suggestions. 
A question regarding my current case, I am using a win 7 dev machine and my target server is Win 2008 server. How would you recommend i approach deploying my application without having to run through the build process? I was thinking to run the build on my local machine and just copy the mem-cached folder to the server. Is that the right approach?

**Comment by 3rd-Eden** [comment]
@morious I can't really advice on that as I'm not Windows user.

**Comment by morious** [comment]
@3rd-Eden solved it by:
1. Build on a local dev machine
2. copy folder to server
3. point to relative path in require statement (instead of installing with npm)

Thx!

**Comment by AleCaste** [comment]
Hello morious.
How did you manage to install the memcached module on Win7?
When I tried to do so, I got an error:
https://github.com/3rd-Eden/memcached/issues/246
